app/controllers/admin/blacklisted_emails_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Admin::BlacklistedEmailsController < ApplicationController
+
+  before_filter :admin_only
+
+  def index
+    @admin_blacklisted_email = AdminBlacklistedEmail.new 
+    if params[:query]
+      @admin_blacklisted_emails = AdminBlacklistedEmail.where(["email LIKE ?", '%' + params[:query] + '%'])
+      @admin_blacklisted_emails = @admin_blacklisted_emails.paginate(page: params[:page], per_page: ArchiveConfig.ITEMS_PER_PAGE)
+    end
+  end
+
+  def new
+    @admin_blacklisted_email = AdminBlacklistedEmail.new
+  end
+
+  def create
+    @admin_blacklisted_email = AdminBlacklistedEmail.new(params[:admin_blacklisted_email])
+
```

Comment:
There is, it's validated for uniqueness in the model and there's a db constraint as well.